hashes/src/hash160.rs

```diff
@@ -88,7 +88,7 @@ mod tests {
             // Hash through high-level API, check hex encoding/decoding
             let hash = hash160::Hash::hash(&test.input[..]);
             assert_eq!(hash, test.output_str.parse::<hash160::Hash>().expect("parse hex"));
-            assert_eq!(&hash[..], &test.output[..]);
+            assert_eq!(&hash[..], &test.output);
```

Comment:
If we add `PartialEq<[u8]> for Hash`, we would also want `PartialEq<Hash> for [u8]`, otherwise you can't write `slice == hash` (but you can do `hash == slice`).

And maybe also `[u8; N]`, but only when `N` is the hash output size. (example: `PartialEq<[u8; 32]> for sha256::Hash`)

Reply:
Yes, that's what I meant by "& co".

Reply:
I like all these ideas. Followup PR though.